packages/langium/test/parser/langium-parser-builder.test.ts

```diff
@@ -437,25 +437,25 @@ describe('Parser calls value converter', () => {
 
     test('Should parse bool correctly', () => {
         expectValue('b true', true);
-        // this is the current 'boolean' behavior when a prop type can't be resolved to just a boolean
-        // either true/undefined, no false in this case
-        expectValue('b false', undefined);
-        // ...then no distinguishing between the bad parse case when the type is unclear
-        expectValue('b asdfg', undefined);
+        expectValue('b false', false);
+        // Any value that cannot be parsed correctly is automatically false
+        expectValue('b asdfg', false);
     });
 
     test('Should parse BigInt correctly', () => {
         expectValue('big 9007199254740991n', BigInt('9007199254740991'));
-        expectValue('big 9007199254740991', undefined);
-        expectValue('big 1.1', undefined);
-        expectValue('big -19458438592374', undefined);
+        // Any value that cannot be parsed correctly is automatically false
+        expectValue('big 9007199254740991', false);
```

Comment:
I would expect a numeric value that cannot be parsed to be `undefined`, rather than `false`. Is this fixed now by the design of the new implementation? If not, I would support reverting this.

Reply:
It's a bug fix, see the explanation above.

Reply:
The boolean case makes sense above, but here less so. It would be a topic I would like to discuss this in a later dev meeting perhaps, and we can move forward with this as is. Just want to be sure that we don't mislead or confuse end-users with how `false` is now a possible value for these property, even when the type is not a boolean.

Reply:
Well, the type is a boolean. The generated type is `value: boolean | number | bigint | string | Date`.

Reply:
Ahhh Right, that's a good point, in this example booleans are part of the prop, and so it would fallback...

So, in cases where `boolean` is not part of the type we should still get `undefined` then as before? I was looking over our cases and don't see any specific tests for checking that we get such a value when the parser fails (but no boolean type for the value). If that's how it works than I'm all good 👍 .

Reply:
Right, I'll add a test for that :)